src/workerd/api/streams/standard.c++

```diff
@@ -2918,57 +2847,41 @@ private:
   }
 };
 
-template <typename T>
-class PumpToReader final: public AllReaderBase {
+class PumpToReader {
 public:
-  using Readable = kj::Own<T>;
-
-  KJ_DISALLOW_COPY_AND_MOVE(PumpToReader);
-
-  PumpToReader(Readable readable, kj::Own<WritableStreamSink> sink, bool end)
+  PumpToReader(jsg::Ref<ReadableStream> stream, kj::Own<WritableStreamSink> sink, bool end)
     : ioContext(IoContext::current()),
-      state(kj::mv(readable)),
-      sink(kj::mv(sink)),
-      end(end) {
-    state.template get<Readable>()->setOwner(addWeakRef());
-  }
+      state(kj::mv(stream)),
+      sink(ioContext.addObject(kj::mv(sink))),
```

Comment:
Can you explain what addObject being added here does?

Reply:
The `sink` is a kj heap i/o object that ends up getting passed into a v8 heap object, which means it needs to be wrapped in an `IoOwn` to be used safely. That's what `addObject` does